react-ui/src/pages/contest/Contest.jsx

```diff
@@ -231,81 +65,46 @@ function Contest() {
-  const [isSettingsOpen, setSettingsOpen] = useState(false);
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const [drawerEntryId, setDrawerEntryId] = useState(null);
   const [votingExpired, setVotingExpired] = useState(false);
-  const [{ votingDisabled }, setComponentsState] = useComponentsState();
 
-  const updateScroll = () => {
-    setScroll({
-      y: window.scrollY,
-    });
-  };
+  // Scroll to top when unmounted
+  // For site-wide solution see https://reactrouter.com/en/main/components/scroll-restoration
+  useEffect(() => () => {
+    scrollInstantlyTo(0);
+  }, []);
 
+  // Check for elements in viewport when isLoaded changes
   useEffect(() => {
     forceCheck();
-  }, [isLoaded]);
+  }, [contest.name]);
 
   useEffect(() => {
-    if (!contest.name) {
-      return;
-    }
-
+    // Clear cache if the voting window is still closed to force fetch again on next visit
     if (contest.votingWindowOpen === false) {
       updateCache(null);
-      setLoaded(true);
-      return;
     }
-
-    const { entryId, y } = scroll;
-    const { innerWidth, requestId, scrollY } = state || {};
-    if (!entryId && !scrollY) {
-      setLoaded(true);
-      return;
-    }
-
-    if (!isLoaded && !scrollingIntervalId) {
-      scrollingIntervalId = setInterval(() => {
-        if (scrollY) {
-          if (window.innerWidth === innerWidth) {
-            scrollInstantlyTo(scrollY);
-          }
-        } else {
-          const entryEl = document.getElementById(entryId);
-          if (!entryEl) {
-            return;
-          }
-
-          let scrollTop = y;
-
-          const headerHeight = document.getElementsByTagName('header')[0].offsetHeight;
-          const { bottom, top } = entryEl.getBoundingClientRect();
-          const windowTop = scrollTop + headerHeight;
-          const windowBottom = scrollTop + window.innerHeight;
-          if (
-            scrollTop === undefined
-            || (bottom < windowTop && top < windowTop)
-            || (bottom > windowBottom && top > windowBottom)
-            || requestId !== contest.requestId
-          ) {
-            scrollTop = top - headerHeight - 8;
-          }
-
-          scrollInstantlyTo(scrollTop);
-        }
-        setLoaded(true);
-        setScroll({});
-        window.history.replaceState({}, document.title);
-        window.history.pushState({ usr: { selectedCategories } }, document.title);
-        clearInterval(scrollingIntervalId);
-        scrollingIntervalId = null;
-      }, 50);
-    }
-  }, [state, contest]);
-
+  }, [contest.votingWindowOpen]);
+
+  // ??? WHAT ELSE CAN BE REMOVED HERE?
+  // useEffect(() => {
+  // instead of isLoaded, I think we can just use contest.name
+  // if (!contest.name) {
+  //   return;
+  // }
+  // const { entryId } = scroll;
+  // const { scrollY } = state || {};
+  // if (!entryId && !scrollY) {
+  //   return;
+  // }
+  // }, [state, contest]);
```

Comment:
Nit: I think you've got the code here working well, we could probably go ahead and delete the comments.

Reply:
Can remove later.